src/models/timeaxisheaderview.cpp

```diff
@@ -0,0 +1,257 @@
+        painter->setPen(QColor(0,0,255));
+        bounding.translate(bounding.width(), 0);
+        bounding.setWidth(prefixWidth);
+        painter->drawText(bounding, Qt::AlignBottom | Qt::AlignLeft, placeholder);
+    }
+
+    for (const auto &tickAndLabel : pfl.ticksAndLabel())
+    {
+        const auto x = xForTime(tickAndLabel.first);
+        if (std::abs(tickAndLabel.first - pfl.prefixValue()) < oneNanoSecond)
+        {
+            painter->setPen(QColor(0,0,0));
+            painter->drawLine(x, rect.y() + fontSize, x, rect.y() + rect.height());
+        }
+        else
+        {
+            painter->setPen(QColor(0,0,255));
+            QRect labelRect(x - endLabelWidth/2, rect.y() + fontSize, endLabelWidth, fontSize);
+            painter->drawText(labelRect, Qt::AlignCenter | Qt::AlignBottom, tickAndLabel.second);
+            painter->drawLine(x, labelRect.y() + fontSize, x, labelRect.y() + rect.height());
+        }
+    }
+}
+    
+
+// Construction
+
+PrefixTickLabels::PrefixTickLabels(const double min, const double max, const unsigned targetNbTicks)
+{
+    const double range = std::abs(max - min);
+    const double mid = (min + max)/2;
+
+    const auto range_10 = static_cast<int>(floor(log10(range))) - (log10(range) <= 0.0 ? 1 : 0);
+
+    const int scale_10 = s_group_10 * static_cast<int>((range_10 - (range_10 < 0 ? s_group_10 - 1 : 0 ))/ s_group_10);
+    const double inv_scale = pow(10, -scale_10);
+
+    m_prefix_10 = scale_10 + s_group_10;
+    const double prefix_power = pow(10, m_prefix_10);
+
+    // Prefer the left most prefix in range
+    std::vector<double> prefixCandidates {
+        ceil(min / prefix_power) * prefix_power,
+        floor(mid / prefix_power) * prefix_power,
+        floor(max / prefix_power) * prefix_power,
+        };
+    std::sort(prefixCandidates.begin(), prefixCandidates.end());
+    const auto prefixCandidate = std::find_if(prefixCandidates.begin(), prefixCandidates.end(), [min, max](const double candidate) {
+        return min <= candidate && candidate <= max;
+    });
+    if (prefixCandidate != prefixCandidates.end())
+    {
+        m_prefixValue = *prefixCandidate;
+    }
+    else
+    {
+        // .. if none found, take the left most
+        m_prefixValue = prefixCandidates.front();
+    }
+
+    const double spacing = niceNum(range / targetNbTicks);
+    const int labelFraction_10 = std::max(0, scale_10 - static_cast<int>(floor(log10(spacing))));
+
+    const double niceMin = ceil(min / spacing) * spacing;
+    const int nbTicks = static_cast<int>(floor((max - niceMin) / spacing));
+
+    const auto siPrefix = PrefixTickLabels::siPrefix(scale_10);
+    m_ticksAndLabels.reserve(nbTicks);
+    for (int j = 0; j <= nbTicks; j++)
+    {
+        const double tick = niceMin + j * spacing;
+        const auto tickLabel = QString(QLatin1String("%1%2")).arg((tick - m_prefixValue) * inv_scale, 0, 'f', labelFraction_10).arg(tick != 0.0 ? siPrefix : QString());
```

Comment:
what is this doing, formatting a fractional number? can't you use a floating point number instead?

Reply:
This is necessary to avoid rounding errors, the second part is a si prefix as can be seen on the screenshot below.